src/buildHTML.js

```diff
@@ -575,12 +579,48 @@ var groupTypes = {
             totalHeight += depth + gap; // \@yargarraycr
             body[r] = outrow;
         }
+
         var offset = totalHeight / 2 + fontMetrics.metrics.axisHeight;
         var coldescriptions = group.value.cols || [];
         var cols = [];
         var colsep;
-        for (c = 0; c < nc; ++c) {
-            var coldescr = coldescriptions[c] || {};
+        var colDescrNum;
+        for (c = 0, colDescrNum = 0;
+             c < nc || colDescrNum < coldescriptions.length;
+             ++c, ++colDescrNum) {
+            var coldescr = coldescriptions[colDescrNum] || {};
+
+            var lastWasSeparator = false;
+            while (coldescr.type === "separator") {
+                if (lastWasSeparator) {
+                    colsep = makeSpan(["arraycolsep"], []);
+                    colsep.style.width = "0.2em";
```

Comment:
Ah ha! I figured it out! (through a lot of guesswork and blind searching through the TeX source).

The expansion of `\array` (which is expanded to when you write \being{array}` eventually calls `\@mkpream` to generate the preamble (which is the analog of the column definitions when using `halign` in plain TeX). In here, it calls `\@testpach`, which looks at the next character, and if said next character is `|`, it sets `\@chclass` to `1`. Then we look at `\@chclass`, and if it is `1` we call `\@classi`, which looks at `\@lastchclass` (presumably the previous `\@chclass`), and if that is also `1`, it adds `\hskip \doublerulesep` before adding the vrule. Observing `\doublerulesep`, we find that it is `2pt`, which ends up being `.2em` with our current `1em=10pt` measurement.

I'll go add this to our font metrics file...